v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -118,10 +117,14 @@ func (ctx KubeGlobalContext) ForTest(t *testing.T) *KubePerTestContext {
 	// Test configs never want SyncPeriod set as it introduces jitter
 	cfg.SyncPeriod = nil
 
-	return ctx.ForTestWithConfig(t, cfg)
+	return ctx.forTestWithConfig(t, cfg, false)
```

Comment:
Bool parameters are difficult to read - suggest changing this so `ForTestWithConfig()` defaults to _included in limit_ but with a method you can call to exclude it.

So this code would become

``` go
return ctx.ForTestWithConfig(t, config).exludeFromConcurrentLimit()
```

Reply:
The issue with doing it this way is that the call to `ctx.createBaseTestContext` in `forTestWithConfig` is the thing that actually creates and launches the envtest. That call is made in `forTestWithConfig` quite early on in the process. Once envtest is launched, none of the tracking stuff can be done. Doing it the way you suggested is difficult because `.excludeFromConcurrentLimit()` gets called after the whole EnvTest launch sequence has already happened.

That's part of the reason why I moved `forTestWithConfig` to be unexported. Callers either use `ForTest`, which implies they're using the default config which is always running, _or_ they use `ForTestWithConfig` which implies they are not using the default config and they get the concurrency limiting and garbage collection I've added. 

So from the callers perspective there is no confusing `bool` parameter, it's hidden. 
I could:
1. Change this to an enum so it reads clearer.
2. Improve the name of the boolean (doesn't really help at the call-site but reads clearer in the internal `forTestWithConfig`).
3. Do a larger refactor to come up with a configuration struct and pass that?

I'm definitely happy to do either 1 or 2... I could probably be convinced to do 3, although I'm not sure it really makes sense as this function only has 3 arguments. 

Thoughts?

Reply:
Ergh. I didn't realise it launched quite so quickly.

Change it to clarify the parameter name with a comment and I'll be happy:

```suggestion
	return ctx.forTestWithConfig(t, cfg, /* countsTowardsLimit: */ false)
```

Reply:
Moved to a simple enum instead for a smidge more clarity.